temboard UI 5.1 - delete instance does nothing

Running:
temboard-5.1-1.el7.noarch

When deleting an instance, nothing happens in the UI:

![image](https://user-images.githubusercontent.com/30243584/83706605-88d16080-a5dd-11ea-86b4-2b19c555665a.png)

The instance to be deleted was upgraded to PG 12 from 9.6. I cannot figure out a better way than to delete the instance from temboard and redeploy it... just to change the PGDATA! **How come we cannot edit PGDATA for an instance?**

The logs show:
```
2020-06-03 21:00:20,907 temboardui[19251]: [access] INFO: 200 GET /json/settings/instance/xxxxxxxxxxx/12254 (xxxxxxxxxxx) 11.97ms
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR: Unhandled Error:
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR: Traceback (most recent call last):
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 224, in error_middleware
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     return func(request, *args)
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 382, in json_middleware
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     return func(request, *args)
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 426, in user_middleware
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     return func(request, *args)
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/handlers/settings/instance.py", line 149, in json_delete_instance
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     delete_instance(request.db_session, **data)
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/application.py", line 452, in delete_instance
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     .filter(MonitoringHost.hostname == instance.hostname) \
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/query.py", line 2640, in count
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     return self.from_self(col).scalar()
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/query.py", line 2426, in scalar
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     ret = self.one()
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/query.py", line 2395, in one
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     ret = list(self)
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/query.py", line 2437, in __iter__
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     self.session._autoflush()
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/session.py", line 1198, in _autoflush
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     self.flush()
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/session.py", line 1919, in flush
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     self._flush(objects)
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/session.py", line 2037, in _flush
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     transaction.rollback(_capture_exception=True)
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib64/python2.7/site-packages/sqlalchemy/util/langhelpers.py", line 60, in __exit__
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     compat.reraise(exc_type, exc_value, exc_tb)
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/session.py", line 2001, in _flush
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     flush_context.execute()
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/unitofwork.py", line 372, in execute
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     rec.execute(self)
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/unitofwork.py", line 479, in execute
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     self.dependency_processor.process_deletes(uow, states)
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/dependency.py", line 530, in process_deletes
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     uowcommit, False)
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/dependency.py", line 575, in _synchronize
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     sync.clear(dest, self.mapper, self.prop.synchronize_pairs)
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:   File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/sync.py", line 57, in clear
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR:     (r, orm_util.state_str(dest))
2020-06-03 21:00:21,889 temboardui[19251]: [web] ERROR: AssertionError: Dependency rule tried to blank-out primary key column 'collector_status.instance_id' on instance '<CollectorStatus at 0x7ff94c0c8110>'
2020-06-03 21:00:21,891 temboardui[19251]: [web] ERROR: Request failed: 500 Dependency rule tried to blank-out primary key column 'collector_status.instance_id' on instance '<CollectorStatus at 0x7ff94c0c8110>'.
2020-06-03 21:00:21,892 temboardui[19251]: [web] ERROR: Unhandled Error:
2020-06-03 21:00:21,892 temboardui[19251]: [web] ERROR: Traceback (most recent call last):
2020-06-03 21:00:21,892 temboardui[19251]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 506, in sync_request_wrapper
2020-06-03 21:00:21,892 temboardui[19251]: [web] ERROR:     return func(request, *args)
2020-06-03 21:00:21,892 temboardui[19251]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 208, in database_middleware
2020-06-03 21:00:21,892 temboardui[19251]: [web] ERROR:     request.db_session.commit()
2020-06-03 21:00:21,892 temboardui[19251]: [web] ERROR:   File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/session.py", line 776, in commit
2020-06-03 21:00:21,892 temboardui[19251]: [web] ERROR:     self.transaction.commit()
2020-06-03 21:00:21,892 temboardui[19251]: [web] ERROR:   File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/session.py", line 375, in commit
2020-06-03 21:00:21,892 temboardui[19251]: [web] ERROR:     self._assert_active(prepared_ok=True)
2020-06-03 21:00:21,892 temboardui[19251]: [web] ERROR:   File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/session.py", line 214, in _assert_active
2020-06-03 21:00:21,892 temboardui[19251]: [web] ERROR:     % self._rollback_exception
2020-06-03 21:00:21,892 temboardui[19251]: [web] ERROR: InvalidRequestError: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: Dependency rule tried to blank-out primary key column 'collector_status.instance_id' on instance '<CollectorStatus at 0x7ff94c0c8110>'
2020-06-03 21:00:21,893 temboardui[19251]: [tornado.general] WARNING: 500 POST /json/settings/delete/instance (10.244.244.145): This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: Dependency rule tried to blank-out primary key column 'collector_status.instance_id' on instance '<CollectorStatus at 0x7ff94c0c8110>'
2020-06-03 21:00:21,893 temboardui[19251]: [access] ERROR: 500 POST /json/settings/delete/instance (10.244.244.145) 16.67ms
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,7 @@
 
 - Full PostgreSQL 12 support, including monitoring and configuration, by [@pgiraud].
 - Double loading of legacy plugins, by [@bersace].
+- Impossibility to delete instance (with SQLAlchemy < 1), by [@pgiraud]
 
 ### Changed
 

--- a/temboardui/plugins/monitoring/model/orm.py
+++ b/temboardui/plugins/monitoring/model/orm.py
@@ -1,7 +1,7 @@
 from temboardui.plugins.monitoring.model import tables
 
 from sqlalchemy.ext.declarative import declarative_base
-from sqlalchemy.orm import relationship
+from sqlalchemy.orm import relationship, backref
 from sqlalchemy.inspection import inspect
 
 
@@ -77,4 +77,6 @@ class CheckState(Model):
 
 class CollectorStatus(Model):
     __table__ = tables.collector_status
-    instance = relationship('Instance', backref='collector_statuts')
+    instance = relationship('Instance',
+                            backref=backref('collector_statuts',
+                                            cascade='all, delete-orphan'))

